### ISSUE TITLE
Build Problem with Svelte and UnoCSS

### ISSUE THREAD

**Issue by Jelenkee** [withastro/astro#5468]
### What version of `astro` are you using?

1.6.11

### Are you using an SSR adapter? If so, which one?

no

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

When I build my website I get an error from UnoCSS.
This only happens when I use the client directive `client:only` (idle and load work fine) and when I use the style tag in the svelte component.

Since I had a similar problem a while ago (#4476) I assume Astro is to blame ;)

### Link to Minimal Reproducible Example

https://stackblitz.com/github/Jelenkee/svelte-bug-astro/tree/main/?file=src%2Fpages%2Findex.astro

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by bluwy** [comment]
```
[unocss:global:build:generate] [unocss] does not found CSS placeholder in the generated chunks
This is likely an internal bug of unocss vite plugin
 error   [unocss] does not found CSS placeholder in the generated chunks
  This is likely an internal bug of unocss vite plugin
```

It says it's a bug within unocss 😄 

Anyways I debugged this and it likely needs a change in `@unocss/astro` to support `client:only`. The problem is that when we do the client build for `client:only` components, there's no `import 'uno.css'` encountered so it logs the error.

it looks like [uno adds that for SSR only](https://github.com/unocss/unocss/blob/22c4e70d53870b9413399e0ed20a7fece610eaa3/packages/astro/src/index.ts#LL59) so the client build fails. Setting it to `page` worked for me but I'm not sure why it's `page-ssr` in the first place.

The other alternative uno could do is to add `uno.css` as a Rollup input during the SSR build but it shouldn't be needed if the above works.

Closing as it's not an Astro issue.
